### ISSUE TITLE
Overloaded functions cannot have keyword arguments

### ISSUE THREAD

**Issue by jez** [sorbet/sorbet#5636]
#### Input

[→ View on sorbet.run](https://sorbet.run/#%23%20typed%3A%20__STDLIB_INTERNAL%0A%0Amodule%20MyKernel%0A%20%20extend%20T%3A%3ASig%0A%20%20sig%20%7Bparams%28x%3A%20T.untyped%29.returns%28Integer%29%7D%0A%20%20sig%20%7Bparams%28x%3A%20T.untyped%2C%20exception%3A%20FalseClass%29.returns%28T.nilable%28Integer%29%29%7D%0A%20%20def%20MyInteger%28x%2C%20exception%3A%20true%29%0A%20%20end%0Aend)

```ruby
# typed: __STDLIB_INTERNAL

module MyKernel
  extend T::Sig
  sig {params(x: T.untyped).returns(Integer)}
  sig {params(x: T.untyped, exception: FalseClass).returns(T.nilable(Integer))}
  def MyInteger(x, exception: true)
  end
end
```

#### Observed output

```
editor.rb:6: Malformed sig. Overloaded functions cannot have keyword arguments:  exception https://srb.help/5003
     6 |  sig {params(x: T.untyped, exception: FalseClass).returns(T.nilable(Integer))}
                                    ^^^^^^^^^
Errors: 1
```

#### Expected behavior

Ideally this functionality would be allowed. It would unblock support for #4336, among other things.

- - -

It used to be complicated internally to be able to support overloaded methods with keyword arguments.

I have a hunch that it might be easier now that a number of improvements have landed in how we support keyword arguments.

**Comment by jez** [comment]
Duplicate of #37.
